### PR TITLE
uart/cdc: Support changing of line coding (uart config)

### LINKED ISSUE #75: cdc: Change UART configuration based on CDC commands
If the client asks for 9600 baud, we should change to that etc.

### PR DESCRIPTION
**Pull request by kbeckmann**
Closes #75

### CODE REVIEW

File: app/drivers/uart.h [hunk 1]
```diff
@@ -15,9 +14,31 @@
 #define EUART_TX						(EUART_BASE + 5)
 #define EUART_RX						(EUART_BASE + 6)
 #define EUART_FLUSH_RX					(EUART_BASE + 7)
+#define EUART_INVALID_ARG				(EUART_BASE + 8)
+
+enum uart_stopbits_t {
```
Comment (arturo182) [inline app/drivers/uart.h]: Why the _t suffix, we don't really use that, for example we have `struct usb_rx_queue_item`, no _t there. isn't _t more for types defined with typedef?

File: app/drivers/usb/cdc.c [hunk 2]
```diff
@@ -26,6 +27,8 @@ static struct {
 	uint8_t rx_queue_storage[QUEUE_LENGTH * QUEUE_ITEM_SIZE];
 	SemaphoreHandle_t tx_done_semaphore;
 	StaticSemaphore_t tx_done_semaphore_buffer;
+
+	struct uart_line_coding_t line_coding;
```
Comment (arturo182) [inline app/drivers/usb/cdc.c]: If the coding is passed to uart, why is this here?

File: app/drivers/uart.c [hunk 3]
```diff
@@ -209,11 +209,48 @@ err_t uart_flush_rx(void)
 	return ERR_OK;
 }
 
-err_t uart_config_set(UART_InitTypeDef *p_config)
+err_t uart_config_set(struct uart_line_coding_t *p_config)
```
Comment (arturo182) [inline app/drivers/uart.c]: const ptr

File: app/drivers/uart.c [hunk 4]
```diff
@@ -209,11 +209,48 @@ err_t uart_flush_rx(void)
...
+	uint32_t value;
+	
+	hal_config.BaudRate = p_config->baudrate_bps;
+
+	if (p_config->databits == 7)
+		value = UART_WORDLENGTH_7B;
+	else if (p_config->databits == 8)
+		value = UART_WORDLENGTH_8B;
+	else
+		return EUART_INVALID_ARG;
```
Comment (arturo182) [inline app/drivers/uart.c]: switch would work too, but this is fine

**Comment by kbeckmann** [comment]
Alright

**Comment by kbeckmann** [comment]
Would be nice if we could get rid of any HAL references from out app code. Now uart.h doesn't need any hal stuff at all. cdc.h still has some.
